### PR TITLE
fix #295675: RNA improvements

### PR DESCRIPTION
**Pull request by MarcSabatella**
Resolves: https://musescore.org/en/node/295675

Most of the rendering for RNA is handled by the Campania font,
so the main part of this change is just including the latest version of that font.
But I also needed to update some of the pre-processing of the RNA elements
to skip the special handling of parentheses and equals sign
(these too are handled directly by Campania),
and to fix a bug in the HTML escaping of angle brackets and other characters.

This PR also includes code to allow the Special Characters palette to function within both RNA and chord symbols, to provide additional flexibility.  The underlying support was there, it just needed to be enabled (and I also fixed a bug in the Special Characters palette for all text).

### CODE REVIEW

**Comment by MarcSabatella** [comment]
Properly handling additional characters that can be used in RNA like angle brackets required me to fix a bug in chord symbols where HTML escaping isn't handled right, so I've added a commit.

Resolves: https://musescore.org/en/node/295778

**Comment by MarcSabatella** [comment]
Another commit added to handle special characters added to chord symbols.  Could potentially have been made a separate PR but the code overlapped just enough to cause merge conflicts.

Resolves: https://musescore.org/en/node/273256 and https://musescore.org/en/node/273266

These changes are only about the user interaction during drap&drop and in edit mode, I don't know a way to automate tests for this.

**Comment by MarcSabatella** [comment]
I've marked this WIP because I probably will make another tweak to Campania, and I do want to test more thoroughly.

**Comment by MarcSabatella** [comment]
I am happy with where this is, so I have removed the WIP label.  Now that there is going to be another RC, I would like us to consider these changes, as they address some limitations/bugs in the original RNA implementation.

File: libmscore/harmony.cpp [hunk 1]
```diff
@@ -634,17 +634,17 @@ const ChordDescription* Harmony::parseHarmony(const QString& ss, int* root, int*
       bool useLiteral = false;
       if (ss.endsWith(' '))
             useLiteral = true;
-      QString s = ss.simplified();
 
       if (_harmonyType == HarmonyType::ROMAN) {
-            _userName = s;
-            _textName = s;
+            _userName = ss;
+            _textName = ss;
```
Comment (dmitrio95) [inline libmscore/harmony.cpp]: Does RNA need to avoid calling `simplified()` on its text? Do consequential spaces make sense there?

Comment (MarcSabatella) [reply libmscore/harmony.cpp]: Yes, it's good for RNA to not simplify, people may well want to embed multiple spaces to get various alignment effects.  That's why I changed this to use "ss", although that change is in the one commit I said I could remove for now if necessary.

Comment (MarcSabatella) [reply libmscore/harmony.cpp]: and btw I have my branch update locally to remove that commit, it works as expected for me, so I could force push it if we decide to skip the special characters improvements

Comment (MarcSabatella) [reply libmscore/harmony.cpp]: Oh, yeah, also - I anticipate supporting mutli-line RNA at some point, that was my original reason to get rid of the simplified.  I had multiline partially working, but underlying issues in handlling of multiline text in TextBase made me stop for now.

**Comment by MarcSabatella** [comment]
BTW, here's the forum thread that prompted this:

https://musescore.org/en/node/295501

Here are some of the enhancements made possible (again, mostly by the font itself):

image

**Pull request merged**